packages/aws-cdk/lib/api/bootstrap/bootstrap-environment2.ts

```diff
@@ -0,0 +1,44 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import * as fs from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+import { BootstrapEnvironmentProps, deployStack, DeployStackResult, ISDK } from '..';
+
+export async function bootstrapEnvironment2(environment: cxapi.Environment, sdk: ISDK,
+                                            toolkitStackName: string, roleArn: string | undefined,
+                                            props: BootstrapEnvironmentProps = {}): Promise<DeployStackResult> {
+  if (props.trustedAccounts?.length && !props.cloudFormationExecutionPolicies?.length) {
+    throw new Error('--cloudformation-execution-policies are required if --trust has been passed!');
```

Comment:
I'm not super happy with this user experience. What if I'm trusting a new account? Not only do I have to list all previous accounts, but I also have to re-list the policy?

And the policy even needs to be referenced by ARN, which is super annoying.

I know we're going bare-bones functionality for now and worry about the experience later, but this is going to need some attention I feel...

Reply:
I agree this is medium.

1. For the "trusting new accounts" problem: would your comment [here](https://github.com/aws/aws-cdk/pull/5856#discussion_r376365024) be applicable? We read the current value of the parameter, and add to it (with set semantics) the array of accounts that were passed in `--trust`, and that's the final value of the parameter? This way, `--trust` would always be additive.

2. For the "policy needs to be referenced by ARN" problem: originally, we wanted to prompt the customer if they didn't pass anything in the `--cloudformation-execution-policies` with a list like this:

```
This will create a new IAM Role, trusting any principal from the ACCOUNT_ID AWS account,
that will be used for deploying your CloudFormation stack(s) in that environment.
Please choose the Managed Policy or Policies to attach to the Role
that contain the permissions required to perform those deployments:

[1]  AdministratorAccess
[2]  AmazonDynamoDBFullAccess
[3]  AmazonEC2FullAccess
[4]  AmazonECS_FullAccess
[5]  AmazonS3FullAccess
[6]  AmazonSageMakerFullAccess
[7]  AmazonSESFullAccess
[8]  AmazonSNSFullAccess
[9]  AmazonSQSFullAccess
[10] AWSLambdaFullAccess
[11] SimpleWorkflowFullAccess
...

Please input the number of the chosen Policy from the above list,
or provide a fully qualified ARN of a different Managed Policy.
You can input multiple numbers and/or ARNs (up to 10),
separated by commas.

>
```

But it was scrapped for time purposes. Do you want to revisit this topic?

Reply:
> 1. For the "trusting new accounts" problem: would your comment here be applicable? We read the current value of the parameter, and add to it (with set semantics) the array of accounts that were passed in --trust, and that's the final value of the parameter? This way, --trust would always be additive.

Yes, I would definitely like that. We would only use `UsePreviousValue` if there was NO change to the parameter, but if we used `--trust` or `--untrust` (?) we would do set logic on the parameter.

2. I think I might want to revisit, or get people to the "right" solutions in an easier way. Maybe a UI, maybe we recognize some aliases for "standard" policies (`"admin" => "arn:aws:iam:::policy/AdministratorAccess"`). Doesn't have to be here and now (let's get this merged), but I want to put it into your head already.

Reply:
> Yes, I would definitely like that. We would only use UsePreviousValue if there was NO change to the parameter, but if we used --trust or --untrust (?) we would do set logic on the parameter.

Blah. This is kind of hard ☹️.

The problem is that you can't use `UsePreviousValue` if you're creating the Stack. Which means it's not good enough for our case. We would have to call `DescribeStack` before the deployment, and based on what that returns:

* If the bootstrap Stack doesn't exist yet (= we're creating it), then pass whatever accounts were given in `--trust` as the value of the `TrustedAccounts` parameter.
* If the bootstrap Stack does exist, add the current value of the `TrustedAccounts` parameter (split on commas of course, as its Type is `CommaSeparatedList`) to the set of accounts passed through `--trust`, and use that as the value.

That also means we need some way to remove accounts from the trust relationship (like the `--untrust` CLI option you mentioned).

Do you want me to add this logic as part of this PR, or are you OK coming back to this later @rix0rrr ?

Reply:
Totally later!